### PR TITLE
feat(countries): Add table schema

### PR DESCRIPTION
**Pull request by AnuragNagpure**
## Description
Backend Table restructure needed including a seeding data file update.    

## Why
make the existing table "countries" multi language capable

Currently the table supports only country names in german and english; however we should be more flexible and also be open for any other language

## Issue

N/A Ref : CPLP-3451

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by Phil91 — COMMENTED** [review]
@jjeroch should only the new table be created without adjusting the existing one?
If we want to adjust the existing one this would result in an adjustment of the existing seeding and queries as well.

**Review by ntruchsess — CHANGES_REQUESTED** [review]
please keep in mind under no circumstances migrations may be created in parallel and then be merged. As the current rc4 contains a migration (see #339) the following conditions must be fulfilled merging this PR:
1) PR #339 must be merged into rc4
2) rc4 must be released and merged into dev
3) The migration that is included in this PR must be reverted (it is crucial to start with the unchanged PortalDBContextSnapshot that comes with the rebase on dev, otherwise the delta that is being calculated is incorrect).
4) this PR must be rebased on dev (to include the migration that is part of PR #339
5) the migration to be included in this PR must be recreated.

File: src/portalbackend/PortalBackend.PortalEntities/Entities/CountriesLongNames.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+using System.ComponentModel.DataAnnotations;
+using System.Text.Json.Serialization;
+
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Entities;
+
+public class CountriesLongNames
```
[the remark this answers is not in the document]
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/CountriesLongNames.cs]: Changed entity name to 'CountryLongName'

File: src/portalbackend/PortalBackend.PortalEntities/Entities/CountriesLongNames.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/********************************************************************************
+ * Copyright (c) 2021, 2023 BMW Group AG
```
[the remark this answers is not in the document]
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/CountriesLongNames.cs]: Removed copyright BMW Group line

File: src/portalbackend/PortalBackend.PortalEntities/PortalDbContext.cs [hunk 3]
```diff
@@ -867,6 +868,20 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                     .Select(e => new ConsentStatus(e))
             );
 
+        modelBuilder.Entity<CountriesLongNames>(entity =>
+               {
+                   entity.HasKey(e => new { e.Alpha2Code, e.ShortName });
+
+                   entity.HasOne(k => k.Language)
+               .WithMany(o => o.CountriesLongNames)
```
[the remark this answers is not in the document]
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.PortalEntities/PortalDbContext.cs]: Formatted properly

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs [hunk 4]
```diff
@@ -1,25 +1,9 @@
...
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- *
- * SPDX-License-Identifier: Apache-2.0
- ********************************************************************************/
-
-// <auto-generated />
+﻿// <auto-generated />
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs:1]: Please readd the copyright header

**Review by Phil91 — COMMENTED** [review]
We should discuss if the old properties in country should be removed or not.

File: src/portalbackend/PortalBackend.PortalEntities/Entities/CountriesLongNames.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/CountriesLongNames.cs]: Its still named `CountryLongNames`

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs (hunk 4 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs]: Header added please check

File: src/portalbackend/PortalBackend.PortalEntities/Entities/CountriesLongNames.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/CountriesLongNames.cs]: fixed

**Comment by ntruchsess** [comment]
> We should discuss if the old properties in country should be removed or not.

good point. There also should be a data-migration script in the migration

File: src/portalbackend/PortalBackend.Migrations/Seeder/Data/country_long_names.json [hunk 5]
```diff
@@ -0,0 +1,2467 @@
...
+    {
+        "alpha2code": "AD",
+        "short_name": "en",   
+        "country_long_name": "Andorra"       
+      },
+    {
+      "alpha2code": "AE",
+      "short_name": "de",  
+      "country_long_name": "United Arab Emirates (the)",
+      "country_name_en": "United Arab Emirates (the)"
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.Migrations/Seeder/Data/country_long_names.json]: done

File: src/portalbackend/PortalBackend.Migrations/Seeder/Data/country_long_names.json [hunk 6]
```diff
@@ -0,0 +1,2467 @@
...
+    {
+      "alpha2code": "NO",
+      "short_name": "en",
+      "country_long_name": "Norway"
+    },
+    {
+      "alpha2code": "NP",
+      "short_name": "de", 
+      "country_long_name": "Nepal",
+      "country_name_en": "Nepal"
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.Migrations/Seeder/Data/country_long_names.json]: done

**Review by ntruchsess — APPROVED** [review]
@Phil91 : I did adjust the naming and removed the old properties from country. Migration has been recreated accordingly. Please double check

**Review by ntruchsess — APPROVED** [review]
@Phil91 : I did adjust the naming and removed the old properties from country adjusting the respective queries in a backward-compatible way. Seeding-data is adjusted as well. Please double-check

**Review by Phil91 — APPROVED** [review]
@ntruchsess Lgtm

**Pull request merged**